design/open/template-server/template-server.md

```diff
@@ -0,0 +1,77 @@
+# Template Server
+
+## About
+
+The template server adds a simple generic templating capability to the
+lakeFS web server.  It adds server-side support for administrators and
+developers to serve templated web objects securely.
+
+Among the planned applications:
+
+* Serve capabilities to clients.  Serving capabilities from a template
+  lets developers add them easily but allows administrators to control
+  what is actually served.
+* Serve prepopulated config files for popular setups.  For instance we
+  can serve `.lakectl` and Hadoop/Spark config files.  These sometimes
+  need to be modified, for instance when lakeFS does not know the name
+  of its endpoint.
+
+  This can be an important part of for time-to-value.  Using templates
+  we offer admins some flexibility in configuration: for instance they
+  might define templates with different values, and clients can select
+  which to receive by path.
+  
+## Template format
+
+We use Golang [text/template][text/template] to expand most templates.
+For browsers (based on filename, e.g. `*.html.tt`) we instead must use
+[html/template][html/template] for safety.
+
+The template is read by lakeFS from a specific (configured) prefix, by
+default `lakefs://lakefs-templates/main`.  IAM authorizes users to see
```

Comment:
I'm not sure what we gain by storing the templates in lakeFS.
This makes it harder for us to create built-in templates.
For example: there is going to be an option to create a new repository pre-configured to be connected to Spark.
So with this suggestion, we will need to pre-upload the template to an existing repository or to the new one, just for the user to get the templated Spark snippet.

@arielshaqed, WDYT?

Reply:
My only concern was that taking templates from lakeFS complicates our first time-to-value tasks - namely showing the user Spark snippets when creating a repository. If @Jonathan-Rosenberg is happy with the current design, I am too.